Sources/DatadogCrashReporting/DDCrashReportingPlugin.swift

```diff
@@ -51,18 +54,27 @@ public class DDCrashReportingPlugin: NSObject, DDCrashReportingPluginType {
             }
         } catch {
             _ = completion(nil)
-            consolePrint("🔥 DatadogCrashReporting error: failed to load crash report: \(error)")
+            consoleLogger?.error("🔥 DatadogCrashReporting error: failed to load crash report", error: error)
         }
     }
 
     public func inject(context: Data) {
         DDCrashReportingPlugin.thirdPartyCrashReporter?.inject(context: context)
     }
-}
 
-// MARK: - Utils
+    public func set(console logger: Logger) {
+        consoleLogger = logger
+    }
 
-/// Function printing `String` content to console.
-internal var consolePrint: (String) -> Void = { content in
-    print(content)
+#if DD_SDK_ENABLE_INTERNAL_MONITORING
+    public func set(internal logger: Logger) {
+        internalLogger = logger
+    }
+#endif
 }
+
+// MARK: - Plugin Logs
+
+internal var consoleLogger: Logger? = nil
+
+internal var internalLogger: Logger? = nil
```

Comment:
Can't find a better way without refactoring the plugin. This pattern is already used in the SDK.